### ISSUE TITLE
Fenced blocks `:::` disable ability to fold sections in R Markdown/Quarto

### ISSUE THREAD

**Issue by GegznaV** [rstudio/rstudio#12851]
### System details

    RStudio Edition : Desktop [Open Source]
    RStudio Version : 2023.3.0.386
    OS Version      : Windows 10 x64 (build 19045)
    R Version       : R version 4.2.2 (2022-10-31 ucrt)

### Steps to reproduce the problem

Instead of 
```r
# Section 1

Something

## Section 1.1

Something
```

write:
```r
# Section 1

:::
Something
:::

## Section 1.1

Something
```

### Describe the problem in detail

Sections after insertion of fenced blocks `:::`, sections lose the ability to be folded (see yellow arrow in the print screen).

![image](https://user-images.githubusercontent.com/12725868/225607756-80a4f35c-894d-4351-8873-cde3c0376cf1.png)

![image](https://user-images.githubusercontent.com/12725868/225607809-529a49ac-ad96-4b46-97be-7aa3ae6a60ed.png)

### Describe the behavior you expected

The same behavior with and without `:::` blocks: headers should fold all the time.

***

Different but possibly related (Is it possible to solve these in one go?):

- #9313

**Comment by ronblum** [comment]
Thanks @GegznaV!

Triage note: This is a regression in Cherry Blossom. It works correctly in RStudio Desktop 2022.12.0+353.

**Comment by mikebessuille** [comment]
@romainfrancois did you do some work in Cherry Blossom that may have affected this?

**Comment by timtmok** [comment]
It might be related to this change: https://github.com/rstudio/rstudio/pull/12172

**Comment by rycalvo** [comment]
> It might be related to this change: #12172

Disabling rainbow fenced divs resolved the issue for me.

**Comment by olivroy** [comment]
This appears fixed in the 2024.10.0+175 version for me on Windows
![image](https://github.com/user-attachments/assets/d4452f55-a084-4bb7-b71b-6c13be97b1c0)

**Comment by gtritchie** [comment]
This reproduced for me in Ocean Storm (2023.12.1), but not in Chocolate Cosmos (2024.04.2) or Cranberry Hibiscus (2024.09.0), so closing as fixed.
